agent/xds/delta.go

```diff
@@ -309,7 +305,7 @@ func (s *Server) processDelta(stream ADSDeltaStream, reqCh <-chan *envoy_discove
 						op.TypeUrl, err)
 				}
 				if sent {
-					sentType[op.TypeUrl] = struct{}{}
+					break // wait until we get an ACK to do more
```

Comment:
What would happen if we send some of those types and then we find one that have an outstanding send? Wouldn't that block envoy from Acking or does the Ack is per Type?

Reply:
Envoy must ACK or NACK each message as part of the protocol. We're simply waiting until it {N,}ACKs everything we send serially. If it does neither then the protocol is busted and you probably need to restart envoy.